File: shopping-cart-backend-java/src/main/java/com/drakeintl/shoppingcart/backend/model/Purchase.java
package com.drakeintl.shoppingcart.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * a request to buy an item
 */
@Data
@Builder
@NoArgsConstructor // framework needs this
@AllArgsConstructor
public class Purchase {

    Long id;
    Long itemId;
    Integer quantity;
}

File: shopping-cart-frontend/src/app/model/purchase.ts
import { Item } from "./item";

export class Purchase {

    id: number = null;
    itemId: number;
    quantity: number;
    item: Item;

    static fromList(items: Item[]): Purchase[] {
        return items.map(i => Purchase.from(i));
    }

    static from(item: Item): Purchase {
        let purchase = new Purchase();
        purchase.itemId = item.id;
        purchase.quantity = item.quantity;
        purchase.item = item;

        return purchase;
    }
}
